File: specs/typespec/org/interviews.go
package org

import (
	"vetchium-api-server.typespec/common"
)

type FeedbackDecision string

type ScheduleInterviewRequest struct {
	CandidacyID               string        `json:"candidacy_id"`
	InterviewType             InterviewType `json:"interview_type"`
	StartsAt                  string        `json:"starts_at"`
	EndsAt                    string        `json:"ends_at"`
	Description               *string       `json:"description,omitempty"`
	InterviewLocation         *string       `json:"interview_location,omitempty"`
	InterviewerEmailAddresses []string      `json:"interviewer_email_addresses"`
}

type ScheduleInterviewResponse struct {
	InterviewID string `json:"interview_id"`
}

type UpdateInterviewRequest struct {
	InterviewID       string  `json:"interview_id"`
	StartsAt          *string `json:"starts_at,omitempty"`
	EndsAt            *string `json:"ends_at,omitempty"`
	Description       *string `json:"description,omitempty"`
	InterviewLocation *string `json:"interview_location,omitempty"`
}

type InterviewIDRequest struct {
	InterviewID string `json:"interview_id"`
}

type AddInterviewerRequest struct {
	InterviewID         string `json:"interview_id"`
	OrgUserEmailAddress string `json:"org_user_email_address"`
}

type RemoveInterviewerRequest struct {
	InterviewID string `json:"interview_id"`
	OrgUserID   string `json:"org_user_id"`
}

type SetInterviewerRSVPRequest struct {
	InterviewID string        `json:"interview_id"`
	RSVP        InterviewRSVP `json:"rsvp"`
}

type SubmitInterviewFeedbackRequest struct {
	InterviewID       string           `json:"interview_id"`
	Decision          FeedbackDecision `json:"decision"`
	Positives         string           `json:"positives"`
	Negatives         string           `json:"negatives"`
	OverallAssessment string           `json:"overall_assessment"`
	CandidateFeedback *string          `json:"candidate_feedback,omitempty"`
}

type InterviewerEntry struct {
	OrgUserID           string         `json:"org_user_id"`
	OrgUserEmailAddress string         `json:"org_user_email_address"`
	DisplayName         string         `json:"display_name"`
	RSVP                *InterviewRSVP `json:"rsvp,omitempty"`
	FeedbackSubmitted   bool           `json:"feedback_submitted"`
}

type InterviewFeedback struct {
	OrgUserID         string           `json:"org_user_id"`
	Decision          FeedbackDecision `json:"decision"`
	Positives         string           `json:"positives"`
	Negatives         string           `json:"negatives"`
	OverallAssessment string           `json:"overall_assessment"`
	CandidateFeedback *string          `json:"candidate_feedback,omitempty"`
	SubmittedAt       string           `json:"submitted_at"`
	UpdatedAt         string           `json:"updated_at"`
}

type FeedbackState string

const (
	FeedbackStateDraft     FeedbackState = "draft"
	FeedbackStateSubmitted FeedbackState = "submitted"
)

// MyInterviewFeedback is the calling interviewer's own feedback (draft or
// submitted) for an interview, used to pre-fill the feedback editor.
type MyInterviewFeedback struct {
	InterviewID       string           `json:"interview_id"`
	State             FeedbackState    `json:"state"`
	Decision          FeedbackDecision `json:"decision"`
	Positives         string           `json:"positives"`
	Negatives         string           `json:"negatives"`
	OverallAssessment string           `json:"overall_assessment"`
	CandidateFeedback *string          `json:"candidate_feedback,omitempty"`
	SubmittedAt       *string          `json:"submitted_at,omitempty"`
	UpdatedAt         string           `json:"updated_at"`
}

type OrgInterview struct {
	InterviewID          string              `json:"interview_id"`
	CandidacyID          string              `json:"candidacy_id"`
	CandidateHandle      string              `json:"candidate_handle"`
	CandidateDisplayName string              `json:"candidate_display_name"`
	OpeningTitle         string              `json:"opening_title"`
	ResumeDownloadURL    string              `json:"resume_download_url"`
	InterviewType        InterviewType       `json:"interview_type"`
	StartsAt             string              `json:"starts_at"`
	EndsAt               string              `json:"ends_at"`
	Description          *string             `json:"description,omitempty"`
	InterviewLocation    *string             `json:"interview_location,omitempty"`
	State                InterviewState      `json:"state"`
	CandidateRSVP        *InterviewRSVP      `json:"candidate_rsvp,omitempty"`
	Interviewers         []InterviewerEntry  `json:"interviewers"`
	Feedback             []InterviewFeedback `json:"feedback"`
}

type ListInterviewsRequest struct {
	FilterCandidacyID  *string          `json:"filter_candidacy_id,omitempty"`
	FilterState        []InterviewState `json:"filter_state,omitempty"`
	FilterStartsAtFrom *string          `json:"filter_starts_at_from,omitempty"`
	FilterStartsAtTo   *string          `json:"filter_starts_at_to,omitempty"`
	PaginationKey      *string          `json:"pagination_key,omitempty"`
	Limit              *int32           `json:"limit,omitempty"`
}

type ListInterviewsResponse struct {
	Interviews        []OrgInterviewSummary `json:"interviews"`
	NextPaginationKey *string               `json:"next_pagination_key,omitempty"`
}

type OrgMyInterview struct {
	InterviewID       string         `json:"interview_id"`
	CandidacyID       string         `json:"candidacy_id"`
	OpeningTitle      string         `json:"opening_title"`
	CandidateName     string         `json:"candidate_name"`
	InterviewType     InterviewType  `json:"interview_type"`
	StartsAt          string         `json:"starts_at"`
	EndsAt            string         `json:"ends_at"`
	State             InterviewState `json:"state"`
	MyRSVP            *InterviewRSVP `json:"my_rsvp,omitempty"`
	FeedbackSubmitted bool           `json:"feedback_submitted"`
}

type ListMyInterviewsRequest struct {
	FilterState   []InterviewState `json:"filter_state,omitempty"`
	PaginationKey *string          `json:"pagination_key,omitempty"`
	Limit         *int32           `json:"limit,omitempty"`
}

type ListMyInterviewsResponse struct {
	Interviews        []OrgMyInterview `json:"interviews"`
	NextPaginationKey *string          `json:"next_pagination_key,omitempty"`
}

// Validation functions
func (r *ScheduleInterviewRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.CandidacyID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "candidacy_id",
			Message: "is required",
		})
	}

	if r.InterviewType != "in_person" && r.InterviewType != "video" && r.InterviewType != "take_home" && r.InterviewType != "other" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_type",
			Message: "must be one of: in_person, video, take_home, other",
		})
	}

	if r.StartsAt == "" {
		errs = append(errs, common.ValidationError{
			Field:   "starts_at",
			Message: "is required",
		})
	}

	if r.EndsAt == "" {
		errs = append(errs, common.ValidationError{
			Field:   "ends_at",
			Message: "is required",
		})
	}

	if r.Description != nil && len(*r.Description) > 2000 {
		errs = append(errs, common.ValidationError{
			Field:   "description",
			Message: "must be at most 2000 characters",
		})
	}

	if r.InterviewLocation != nil && len(*r.InterviewLocation) > 2000 {
		errs = append(errs, common.ValidationError{
			Field:   "interview_location",
			Message: "must be at most 2000 characters",
		})
	}

	if len(r.InterviewerEmailAddresses) < 1 || len(r.InterviewerEmailAddresses) > 5 {
		errs = append(errs, common.ValidationError{
			Field:   "interviewer_email_addresses",
			Message: "must have 1-5 items",
		})
	}

	return errs
}

func (r *UpdateInterviewRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_id",
			Message: "is required",
		})
	}

	if r.Description != nil && len(*r.Description) > 2000 {
		errs = append(errs, common.ValidationError{
			Field:   "description",
			Message: "must be at most 2000 characters",
		})
	}

	if r.InterviewLocation != nil && len(*r.InterviewLocation) > 2000 {
		errs = append(errs, common.ValidationError{
			Field:   "interview_location",
			Message: "must be at most 2000 characters",
		})
	}

	return errs
}

func (r *InterviewIDRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_id",
			Message: "is required",
		})
	}

	return errs
}

func (r *AddInterviewerRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_id",
			Message: "is required",
		})
	}

	if r.OrgUserEmailAddress == "" {
		errs = append(errs, common.ValidationError{
			Field:   "org_user_email_address",
			Message: "is required",
		})
	}

	return errs
}

func (r *RemoveInterviewerRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_id",
			Message: "is required",
		})
	}

	if r.OrgUserID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "org_user_id",
			Message: "is required",
		})
	}

	return errs
}

func (r *SetInterviewerRSVPRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_id",
			Message: "is required",
		})
	}

	if r.RSVP != "yes" && r.RSVP != "no" {
		errs = append(errs, common.ValidationError{
			Field:   "rsvp",
			Message: "must be one of: yes, no",
		})
	}

	return errs
}

func (r *SubmitInterviewFeedbackRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{
			Field:   "interview_id",
			Message: "is required",
		})
	}

	if r.Decision != "strong_yes" && r.Decision != "yes" && r.Decision != "neutral" && r.Decision != "no" && r.Decision != "strong_no" {
		errs = append(errs, common.ValidationError{
			Field:   "decision",
			Message: "must be one of: strong_yes, yes, neutral, no, strong_no",
		})
	}

	if r.Positives == "" || len(r.Positives) < 1 || len(r.Positives) > 4000 {
		errs = append(errs, common.ValidationError{
			Field:   "positives",
			Message: "must be between 1 and 4000 characters",
		})
	}

	if r.Negatives == "" || len(r.Negatives) < 1 || len(r.Negatives) > 4000 {
		errs = append(errs, common.ValidationError{
			Field:   "negatives",
			Message: "must be between 1 and 4000 characters",
		})
	}

	if r.OverallAssessment == "" || len(r.OverallAssessment) < 1 || len(r.OverallAssessment) > 4000 {
		errs = append(errs, common.ValidationError{
			Field:   "overall_assessment",
			Message: "must be between 1 and 4000 characters",
		})
	}

	if r.CandidateFeedback != nil && len(*r.CandidateFeedback) > 2000 {
		errs = append(errs, common.ValidationError{
			Field:   "candidate_feedback",
			Message: "must be at most 2000 characters",
		})
	}

	return errs
}

// ValidateDraft is the lenient validation used by save-interview-feedback: a
// draft may have empty positives/negatives/overall_assessment (work in
// progress); only the interview_id, decision and upper bounds are enforced.
func (r *SubmitInterviewFeedbackRequest) ValidateDraft() []common.ValidationError {
	var errs []common.ValidationError

	if r.InterviewID == "" {
		errs = append(errs, common.ValidationError{Field: "interview_id", Message: "is required"})
	}
	if r.Decision != "strong_yes" && r.Decision != "yes" && r.Decision != "neutral" && r.Decision != "no" && r.Decision != "strong_no" {
		errs = append(errs, common.ValidationError{
			Field:   "decision",
			Message: "must be one of: strong_yes, yes, neutral, no, strong_no",
		})
	}
	if len(r.Positives) > 4000 {
		errs = append(errs, common.ValidationError{Field: "positives", Message: "must be at most 4000 characters"})
	}
	if len(r.Negatives) > 4000 {
		errs = append(errs, common.ValidationError{Field: "negatives", Message: "must be at most 4000 characters"})
	}
	if len(r.OverallAssessment) > 4000 {
		errs = append(errs, common.ValidationError{Field: "overall_assessment", Message: "must be at most 4000 characters"})
	}
	if r.CandidateFeedback != nil && len(*r.CandidateFeedback) > 2000 {
		errs = append(errs, common.ValidationError{Field: "candidate_feedback", Message: "must be at most 2000 characters"})
	}
	return errs
}

func (r *ListInterviewsRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.Limit != nil && (*r.Limit < 1 || *r.Limit > 100) {
		errs = append(errs, common.ValidationError{
			Field:   "limit",
			Message: "must be between 1 and 100",
		})
	}

	return errs
}

func (r *ListMyInterviewsRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.Limit != nil && (*r.Limit < 1 || *r.Limit > 100) {
		errs = append(errs, common.ValidationError{
			Field:   "limit",
			Message: "must be between 1 and 100",
		})
	}

	return errs
}

File: specs/typespec/org/interviews.ts
import type {
	InterviewType,
	InterviewState,
	InterviewRSVP,
} from "../hub/candidacies.js";
import type { OrgInterviewSummary } from "./candidacies.js";

export type FeedbackDecision =
	| "strong_yes"
	| "yes"
	| "neutral"
	| "no"
	| "strong_no";

export interface ScheduleInterviewRequest {
	candidacy_id: string;
	interview_type: InterviewType;
	starts_at: string;
	ends_at: string;
	description?: string;
	/** Free-text physical address or video-meeting link (0..2000). */
	interview_location?: string;
	interviewer_email_addresses: string[];
}

export interface ScheduleInterviewResponse {
	interview_id: string;
}

export interface UpdateInterviewRequest {
	interview_id: string;
	starts_at?: string;
	ends_at?: string;
	description?: string;
	interview_location?: string;
}

export interface InterviewIdRequest {
	interview_id: string;
}

export interface AddInterviewerRequest {
	interview_id: string;
	org_user_email_address: string;
}

export interface RemoveInterviewerRequest {
	interview_id: string;
	org_user_id: string;
}

export interface SetInterviewerRSVPRequest {
	interview_id: string;
	rsvp: InterviewRSVP;
}

export interface SubmitInterviewFeedbackRequest {
	interview_id: string;
	decision: FeedbackDecision;
	positives: string;
	negatives: string;
	overall_assessment: string;
	candidate_feedback?: string;
}

export interface InterviewerEntry {
	org_user_id: string;
	org_user_email_address: string;
	display_name: string;
	rsvp?: InterviewRSVP;
	feedback_submitted: boolean;
}

export interface InterviewFeedback {
	org_user_id: string;
	decision: FeedbackDecision;
	positives: string;
	negatives: string;
	overall_assessment: string;
	candidate_feedback?: string;
	submitted_at: string;
	updated_at: string;
}

export type FeedbackState = "draft" | "submitted";

/** The calling interviewer's own feedback (draft or submitted) for an interview. */
export interface MyInterviewFeedback {
	interview_id: string;
	state: FeedbackState;
	decision: FeedbackDecision;
	positives: string;
	negatives: string;
	overall_assessment: string;
	candidate_feedback?: string;
	submitted_at?: string;
	updated_at: string;
}

export interface OrgInterview {
	interview_id: string;
	candidacy_id: string;
	interview_type: InterviewType;
	starts_at: string;
	ends_at: string;
	description?: string;
	interview_location?: string;
	state: InterviewState;
	candidate_rsvp?: InterviewRSVP;
	interviewers: InterviewerEntry[];
	feedback: InterviewFeedback[];
}

export interface ListInterviewsRequest {
	filter_candidacy_id?: string;
	filter_state?: InterviewState[];
	filter_starts_at_from?: string;
	filter_starts_at_to?: string;
	pagination_key?: string;
	limit?: number;
}

export interface ListInterviewsResponse {
	interviews: OrgInterviewSummary[];
	next_pagination_key?: string;
}

export interface OrgMyInterview {
	interview_id: string;
	candidacy_id: string;
	opening_title: string;
	candidate_name: string;
	interview_type: InterviewType;
	starts_at: string;
	ends_at: string;
	state: InterviewState;
	my_rsvp?: InterviewRSVP;
	feedback_submitted: boolean;
}

export interface ListMyInterviewsRequest {
	filter_state?: InterviewState[];
	pagination_key?: string;
	limit?: number;
}

export interface ListMyInterviewsResponse {
	interviews: OrgMyInterview[];
	next_pagination_key?: string;
}

export interface ValidationError {
	field: string;
	message: string;
}

export function validateScheduleInterviewRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.candidacy_id !== "string" || r.candidacy_id.trim() === "") {
		errors.push({
			field: "candidacy_id",
			message: "Must be a non-empty string",
		});
	}

	if (!isValidInterviewType(r.interview_type)) {
		errors.push({
			field: "interview_type",
			message: "Must be one of: in_person, video, take_home, other",
		});
	}

	if (typeof r.starts_at !== "string") {
		errors.push({ field: "starts_at", message: "Must be a valid datetime" });
	}

	if (typeof r.ends_at !== "string") {
		errors.push({ field: "ends_at", message: "Must be a valid datetime" });
	}

	if (r.description !== undefined) {
		if (typeof r.description !== "string") {
			errors.push({ field: "description", message: "Must be a string" });
		} else if (r.description.length > 2000) {
			errors.push({
				field: "description",
				message: "Must be at most 2000 characters",
			});
		}
	}

	if (r.interview_location !== undefined) {
		if (typeof r.interview_location !== "string") {
			errors.push({ field: "interview_location", message: "Must be a string" });
		} else if (r.interview_location.length > 2000) {
			errors.push({
				field: "interview_location",
				message: "Must be at most 2000 characters",
			});
		}
	}

	if (!Array.isArray(r.interviewer_email_addresses)) {
		errors.push({
			field: "interviewer_email_addresses",
			message: "Must be an array",
		});
	} else {
		if (
			r.interviewer_email_addresses.length < 1 ||
			r.interviewer_email_addresses.length > 5
		) {
			errors.push({
				field: "interviewer_email_addresses",
				message: "Must have 1-5 items",
			});
		}
		for (const email of r.interviewer_email_addresses) {
			if (typeof email !== "string" || email.trim() === "") {
				errors.push({
					field: "interviewer_email_addresses",
					message: "All items must be non-empty strings",
				});
				break;
			}
		}
	}

	return errors;
}

export function validateUpdateInterviewRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.interview_id !== "string" || r.interview_id.trim() === "") {
		errors.push({
			field: "interview_id",
			message: "Must be a non-empty string",
		});
	}

	if (r.starts_at !== undefined && typeof r.starts_at !== "string") {
		errors.push({ field: "starts_at", message: "Must be a valid datetime" });
	}

	if (r.ends_at !== undefined && typeof r.ends_at !== "string") {
		errors.push({ field: "ends_at", message: "Must be a valid datetime" });
	}

	if (r.description !== undefined) {
		if (typeof r.description !== "string") {
			errors.push({ field: "description", message: "Must be a string" });
		} else if (r.description.length > 2000) {
			errors.push({
				field: "description",
				message: "Must be at most 2000 characters",
			});
		}
	}

	if (r.interview_location !== undefined) {
		if (typeof r.interview_location !== "string") {
			errors.push({ field: "interview_location", message: "Must be a string" });
		} else if (r.interview_location.length > 2000) {
			errors.push({
				field: "interview_location",
				message: "Must be at most 2000 characters",
			});
		}
	}

	return errors;
}

export function validateInterviewIdRequest(req: unknown): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.interview_id !== "string" || r.interview_id.trim() === "") {
		errors.push({
			field: "interview_id",
			message: "Must be a non-empty string",
		});
	}

	return errors;
}

export function validateAddInterviewerRequest(req: unknown): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.interview_id !== "string" || r.interview_id.trim() === "") {
		errors.push({
			field: "interview_id",
			message: "Must be a non-empty string",
		});
	}

	if (
		typeof r.org_user_email_address !== "string" ||
		r.org_user_email_address.trim() === ""
	) {
		errors.push({
			field: "org_user_email_address",
			message: "Must be a non-empty string",
		});
	}

	return errors;
}

export function validateRemoveInterviewerRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.interview_id !== "string" || r.interview_id.trim() === "") {
		errors.push({
			field: "interview_id",
			message: "Must be a non-empty string",
		});
	}

	if (typeof r.org_user_id !== "string" || r.org_user_id.trim() === "") {
		errors.push({
			field: "org_user_id",
			message: "Must be a non-empty string",
		});
	}

	return errors;
}

export function validateSetInterviewerRSVPRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.interview_id !== "string" || r.interview_id.trim() === "") {
		errors.push({
			field: "interview_id",
			message: "Must be a non-empty string",
		});
	}

	if (!isValidInterviewRSVP(r.rsvp)) {
		errors.push({ field: "rsvp", message: "Must be one of: yes, no" });
	}

	return errors;
}

export function validateSubmitInterviewFeedbackRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.interview_id !== "string" || r.interview_id.trim() === "") {
		errors.push({
			field: "interview_id",
			message: "Must be a non-empty string",
		});
	}

	if (!isValidFeedbackDecision(r.decision)) {
		errors.push({
			field: "decision",
			message: "Must be one of: strong_yes, yes, neutral, no, strong_no",
		});
	}

	if (typeof r.positives !== "string") {
		errors.push({ field: "positives", message: "Must be a string" });
	} else if (r.positives.length < 1 || r.positives.length > 4000) {
		errors.push({
			field: "positives",
			message: "Must be between 1 and 4000 characters",
		});
	}

	if (typeof r.negatives !== "string") {
		errors.push({ field: "negatives", message: "Must be a string" });
	} else if (r.negatives.length < 1 || r.negatives.length > 4000) {
		errors.push({
			field: "negatives",
			message: "Must be between 1 and 4000 characters",
		});
	}

	if (typeof r.overall_assessment !== "string") {
		errors.push({ field: "overall_assessment", message: "Must be a string" });
	} else if (
		r.overall_assessment.length < 1 ||
		r.overall_assessment.length > 4000
	) {
		errors.push({
			field: "overall_assessment",
			message: "Must be between 1 and 4000 characters",
		});
	}

	if (r.candidate_feedback !== undefined) {
		if (typeof r.candidate_feedback !== "string") {
			errors.push({ field: "candidate_feedback", message: "Must be a string" });
		} else if (r.candidate_feedback.length > 2000) {
			errors.push({
				field: "candidate_feedback",
				message: "Must be at most 2000 characters",
			});
		}
	}

	return errors;
}

export function validateListInterviewsRequest(req: unknown): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (r.limit !== undefined) {
		if (typeof r.limit !== "number" || r.limit < 1 || r.limit > 100) {
			errors.push({ field: "limit", message: "Must be between 1 and 100" });
		}
	}

	if (r.pagination_key !== undefined && typeof r.pagination_key !== "string") {
		errors.push({ field: "pagination_key", message: "Must be a string" });
	}

	return errors;
}

export function validateListMyInterviewsRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (r.limit !== undefined) {
		if (typeof r.limit !== "number" || r.limit < 1 || r.limit > 100) {
			errors.push({ field: "limit", message: "Must be between 1 and 100" });
		}
	}

	if (r.pagination_key !== undefined && typeof r.pagination_key !== "string") {
		errors.push({ field: "pagination_key", message: "Must be a string" });
	}

	return errors;
}

function isValidInterviewType(value: unknown): value is InterviewType {
	return (
		value === "in_person" ||
		value === "video" ||
		value === "take_home" ||
		value === "other"
	);
}

function isValidInterviewRSVP(value: unknown): value is InterviewRSVP {
	return value === "yes" || value === "no";
}

function isValidFeedbackDecision(value: unknown): value is FeedbackDecision {
	return (
		value === "strong_yes" ||
		value === "yes" ||
		value === "neutral" ||
		value === "no" ||
		value === "strong_no"
	);
}
